packages/cdk/lib/cloudwatch-logs-management.ts

```diff
@@ -1,11 +1,214 @@
+				RETENTION_IN_DAYS: retentionInDays.toString(),
+			},
+			timeout: Duration.minutes(1),
+		});
+
+		this.overrideLogicalId(setRetentionLambda, {
+			logicalId: 'SetRetentionFunc',
+			reason: 'Migrating from YAML',
+		});
+
+		const setRetentionPolicy = new ManagedPolicy(this, 'SetRetentionPolicy', {
+			statements: [
+				new PolicyStatement({
+					effect: Effect.ALLOW,
+					actions: ['logs:DescribeLogGroups', 'logs:PutRetentionPolicy'],
+					resources: [`arn:aws:logs:${region}:${account}:log-group:*`],
+				}),
+			],
+		});
+		setRetentionLambda.role?.addManagedPolicy(setRetentionPolicy);
+
+		const shipLogEntriesLambda = new GuLambdaFunction(
+			this,
+			'ship-log-entries',
+			{
+				app: 'ship-log-entries',
+				runtime: Runtime.NODEJS_16_X,
+				fileName: 'cloudwatch-logs-management.zip',
+				handler: 'shipLogEntries.shipLogEntries',
+				timeout: Duration.seconds(5),
+				environment: {
+					LOG_KINESIS_STREAM: kinesisStreamArn,
+					STRUCTURED_DATA_BUCKET: structuredFieldsBucket.bucketName,
+				},
+			},
+		);
+
+		this.overrideLogicalId(shipLogEntriesLambda, {
+			logicalId: 'ShipLogEntriesFunc',
+			reason: 'Migrating from YAML',
+		});
+
+		shipLogEntriesLambda.addPermission('ShipLogEntriesPermission', {
+			principal: new ServicePrincipal(`logs.${region}.amazonaws.com`),
+			sourceAccount: this.account,
+		});
+
+		const shipLogEntriesPolicies = [
+			new ManagedPolicy(this, 'ShipLogEntriesPolicy', {
+				statements: [
+					new PolicyStatement({
+						effect: Effect.ALLOW,
+						actions: ['kinesis:PutRecords'],
+						resources: [kinesisStreamArn],
+					}),
+					new PolicyStatement({
+						effect: Effect.ALLOW,
+						actions: ['s3:GetObject'],
+						resources: [`${structuredFieldsBucket.bucketArn}/*`],
+					}),
+				],
+			}),
+			new ManagedPolicy(this, 'DisableCloudWatchLoggingPolicy', {
+				statements: [
+					new PolicyStatement({
+						effect: Effect.DENY,
+						actions: [
+							'logs:CreateLogGroup',
+							'logs:CreateLogStream',
+							'logs:PutLogEvents',
+						],
+						resources: [`arn:aws:logs:*:*:*`],
+					}),
+				],
+			}),
+		];
+
+		shipLogEntriesPolicies.forEach((policy) =>
+			shipLogEntriesLambda.role?.addManagedPolicy(policy),
+		);
+
+		const setLogShippingLambda = new GuScheduledLambda(
+			this,
+			'set-log-shipping',
+			{
+				app: 'set-log-shipping',
+				runtime: Runtime.NODEJS_16_X,
+				fileName: 'cloudwatch-logs-management.zip',
+				handler: 'app.setLogShipping',
+				rules: [{ schedule: Schedule.rate(Duration.minutes(10)) }],
+				monitoringConfiguration: { noMonitoring: true },
+				environment: {
+					LOG_SHIPPING_LAMBDA_ARN: shipLogEntriesLambda.functionArn,
+					LOG_NAME_PREFIXES: logShippingPrefixes.join(','),
+					STRUCTURED_DATA_BUCKET: structuredFieldsBucket.bucketName,
+				},
+				timeout: Duration.minutes(1),
+			},
+		);
+
+		this.overrideLogicalId(setLogShippingLambda, {
+			logicalId: 'SetLogShippingFunc',
+			reason: 'Migrating from YAML',
+		});
```

Comment:
It should be safe to allow CDK to generate the logical ID for the three lambdas as there is no state held in them. I kept it here to make the change set a little smaller/easier to reason about. I propose we move the overrides in a follow-up PR.

Reply:
> I kept it here to make the change set a little smaller/easier to reason about.

👍 

To be honest I'm not 100% sure what will happen if we replace the `shipLogEntriesLambda`... I would probably just leave these overrides in place permanently.